extensions/observability-devservices/deployment/src/main/java/io/quarkus/observability/deployment/ObservabilityDevServiceProcessor.java

```diff
@@ -250,14 +257,24 @@ public DevServicesResultBuildItem.RunningDevService get() {
                 .locateContainer(
                         capturedDevServicesConfiguration.serviceName(), capturedDevServicesConfiguration.shared(),
                         LaunchMode.current(), (p, ca) -> config.putAll(dev.config(p, ca.getHost(), ca.getPort())))
-                .map(new Function<String, DevServicesResultBuildItem.RunningDevService>() {
-                    @Override
-                    public DevServicesResultBuildItem.RunningDevService apply(String cid) {
-                        log.infof("Dev Service %s re-used, config: %s", devId, config);
-                        return new DevServicesResultBuildItem.RunningDevService(Feature.OBSERVABILITY.getName(), cid,
-                                null, config);
-                    }
+                .map(cid -> {
+                    log.infof("Dev Service %s re-used, config: %s", devId, config);
+                    return new DevServicesResultBuildItem.RunningDevService(Feature.OBSERVABILITY.getName(), cid,
+                            null, config);
                 })
+                .or(() -> ComposeLocator.locateContainer(composeProjectBuildItem,
+                        List.of(capturedDevServicesConfiguration.imageName()), LaunchMode.current())
```

Comment:
@ozangunalp so no need for additional label / tag? Image name is enough?

Reply:
If observability service can be anything, it is better to specify the image-name directly or use the container locator label.

Reply:
It can be multiple services, but should have unique image-name.